corehq/motech/value_source.py

```diff
@@ -377,36 +418,22 @@ def _get_commcare_value(self, case_trigger_info):
             return get_ancestor_location_metadata_value(location, self.location_field)
 
 
-class JsonPathMixin(DocumentSchema):
-    """
-    Used for importing a value from a JSON document.
-    """
-    jsonpath = StringProperty(required=True, validators=not_blank)
-
-    def _get_external_value(self, external_data):
-        jsonpath = parse_jsonpath(self.jsonpath)
-        matches = jsonpath.find(external_data)
-        values = [m.value for m in matches]
-        if not values:
-            return None
-        elif len(values) == 1:
-            return values[0]
-        else:
-            return values
-
-    def get_import_value(self, external_data):
-        external_value = self._get_external_value(external_data)
-        return self.deserialize(external_value)
-
-
 class JsonPathCaseProperty(CaseProperty, JsonPathMixin):
-    pass
+
+    def __init__(self, *args, **kwargs):
+        DocumentSchema.__init__(self, *args, **kwargs)
```

Comment:
Why not use `super()`? This is fragile: if `JsonPathMixin` ever gets an `__init__` method it will not be called unless this is changed to `super().__init__(*args, **kw)`.

Reply:
`JsonPathCasePropertyMap` extends `CasePropertyMap`, and both are deprecated, so you when you instantiate `JsonPathCasePropertyMap` you get a deprecation warning for `CasePropertyMap` too. This felt wrong (although maybe it isn't) so I changed it to call `DocumentSchema.__init__()` instead of `super().__init__()`. ... And then applied the same change to all deprecated classes, which might not have been reasonable.

I agree this change feels wrong too. 

I'll change to `super()`, and mentally bump the priority for dropping these classes.